### PR TITLE
FSET-1629 Added event description to faststream csv

### PR DESCRIPTION
**Pull request by WarFox**

### CODE REVIEW

File: app/model/persisted/eventschedules/Event.scala [hunk 1]
```diff
@@ -25,6 +25,7 @@ import reactivemongo.bson.Macros
 
 case class Event(id: String,
                  eventType: EventType,
+                 eventDescription: String,
```
Comment (fayimora) [inline app/model/persisted/eventschedules/Event.scala]: Let's call this `description`.

File: app/services/events/EventsParsingService.scala [hunk 2]
```diff
@@ -38,11 +38,11 @@ object EventsParsingService extends EventsParsingService {
...
-    "CHAIR" -> 10,
-    "DEPARTMENTAL_ASSESSOR" -> 11,
-    "EXERCISE_MARKER" -> 12,
-    "QUALITY_ASSURANCE_COORDINATOR" -> 13
+    "ASSESSOR" -> 10,
+    "CHAIR" -> 11,
+    "DEPARTMENTAL_ASSESSOR" -> 12,
+    "EXERCISE_MARKER" -> 13,
+    "QUALITY_ASSURANCE_COORDINATOR" -> 14
   )
```
Comment (fayimora) [inline app/services/events/EventsParsingService.scala]: Aren't we adding `Sifter` anymore? Confirm from Al

File: app/services/events/EventsParsingService.scala [hunk 3]
```diff
@@ -97,4 +69,46 @@ trait EventsParsingService {
       }
     }
   }
+
+  lazy val df = DateTimeFormat.forPattern("HH:mm")
+
+  private def event(items: String): Event = {
```
Comment (fayimora) [inline app/services/events/EventsParsingService.scala]: I think you should merge this into `event(items: Array[String])` below

File: app/services/events/EventsParsingService.scala [hunk 4]
```diff
@@ -97,4 +69,46 @@ trait EventsParsingService {
...
+    event(items.split(", ?", -1))
+  }
+
+  private def event(items: Array[String]): Event = {
```
Comment (fayimora) [inline app/services/events/EventsParsingService.scala]: Following my comment above, this function will take a `String` and return an `Event` so perhaps the name of the function should reflect that somehow.

File: app/services/events/EventsParsingService.scala [hunk 5]
```diff
@@ -97,4 +69,46 @@ trait EventsParsingService {
...
+    val location = items(2)
+    val venue = VenueType.withName(items(3).replaceAll("\\s|-", "_").toUpperCase)
+    val date = LocalDate.parse(items(4), DateTimeFormat.forPattern("dd/MM/yy"))
+    val startTime = df.parseLocalTime(items(5))
+    val endTime = df.parseLocalTime(items(6))
+    val capacity = items(7).toInt
+    val minViableAttendees = items(8).toInt
+    val attendeeSafetyMargin = items(9).toInt
+
+    if(eventDescription.length > 10) throw new Exception("Event description cannot be more than 10 characters")
```
Comment (fayimora) [inline app/services/events/EventsParsingService.scala]: This restriction is not reported to a user so perhaps we want to get a `slice()` here? Check with Alastair

I think it'll be pretty annoying if we have 30 events and 25 of them have description > 10chars. Also, 10 chars seems a bit short for a description. Check with Alastair

File: it/repositories/EventsRepositorySpec.scala [hunk 6]
```diff
@@ -10,25 +10,25 @@ class EventsRepositorySpec extends MongoRepositorySpec {
   override val collectionName: String = CollectionNames.ASSESSMENT_EVENTS
   lazy val repository = repositories.eventsRepository
   val events = List(
-    Event(id = UUIDFactory.generateUUID(), eventType = EventType.FSAC, location = "London",
+    Event(id = UUIDFactory.generateUUID(), eventDescription = "SAMPLE", eventType = EventType.FSAC, location = "London",
```
Comment (fayimora) [inline it/repositories/EventsRepositorySpec.scala]: Minor nit but perhaps you want to use a real description here.

File: test/services/events/EventsParsingServiceSpec.scala [hunk 7]
```diff
@@ -34,24 +34,41 @@ class EventsParsingServiceSpec extends BaseServiceSpec {
     "fails gracefully when any field is malformed" in new MalformedTestFixture {
       an[Exception] must be thrownBy service.processCentres().futureValue
     }
-  }
 
+    "fails gracefully when event description is longer than 10 malformed" in new LongEventDescription {
+      an[Exception] must be thrownBy service.processCentres().futureValue
+      the[Exception] thrownBy service.processCentres().futureValue must have message "The future returned an exception " +
+        "of type: java.lang.Exception, with message: Error on L3 of the CSV. Event description cannot be more than 10 " +
+        "characters. java.lang.Exception."
```
Comment (fayimora) [inline test/services/events/EventsParsingServiceSpec.scala]: Not sure if you want to test against the message here. #BrittleTests

**Comment by kjharkin** [comment]
Looks good.

**Pull request merged**